tests/ExampleApps/DrawPrimitivesApp/CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+set(app_target testDrawPrimitives)
+
+# Qt stuff //////////////////////
+
+find_package(OpenGL     REQUIRED)
+find_package(Qt5 COMPONENTS Core Widgets OpenGL REQUIRED)
+
+set( Qt5_LIBRARIES
+     Qt5::Core
+     Qt5::Widgets
+     Qt5::OpenGL )
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+
+if(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} VERSION_GREATER "3.9")
+    cmake_policy(SET CMP0071 NEW)
+endif()
+
+# ///////////////////////////////
+
+# Access to Radium headers and declarations/defintions
+include_directories(
+    .
+    ${RADIUM_INCLUDE_DIRS}
+)
+
+
+
+# Get files
+file( GLOB file_sources *.cpp *.c )
+file( GLOB file_headers *.hpp *.h )
+
+
+# Generate an executable
+add_executable( ${app_target} ${file_sources} ${file_headers} )
+
+add_dependencies( ${app_target} radiumEngine radiumGuiBase radiumCore radium_assets)
+
+
+# Link good libraries
+target_link_libraries( ${app_target} # target
+    ${RADIUM_LIBRARIES}              # Radium libs
+    ${GLBINDING_LIBRARIES}           # Radium dep
+    ${Qt5_LIBRARIES}                 # the Qt beast
+)
+
+if (MSVC)
+    set_property( TARGET ${app_target} PROPERTY IMPORTED_LOCATION "${RADIUM_BINARY_OUTPUT_PATH}" )
+endif(MSVC)
+
+radium_cotire( ${app_target} )
```

Comment:
This is not the good way to configure an application (even an example one). To have relocatable app under MacOs, Windows and Linux, and to respect the new build chain philosophy, you must follow the application guideline for cmakefile. Here is a template you can use (assuming the `find_package(Radium REQUIRED Core Engine GuiBase PluginBase IO) is done in a parent directory) : 
```cmake
# Qt stuff //////////////////////
find_package(Qt5 COMPONENTS Core Widgets OpenGL REQUIRED)
set(Qt5_LIBRARIES Qt5::Core Qt5::Widgets Qt5::OpenGL)
set(CMAKE_AUTOMOC ON)
set(CMAKE_AUTORCC ON)
set(CMAKE_INCLUDE_CURRENT_DIR ON)

# Application specific configuration
set(app_sources
  list here the .cpp files
)
set(app_headers
  ...
)
set(app_inlines
  ...
)
set(app_uis
  ...
)
qt5_wrap_ui(app_uis_moc ${app_uis})

set(app_resources
  ... Data files, shaders, ...
)

include_directories(
    ${RADIUM_INCLUDE_DIRS}
    ${CMAKE_CURRENT_BINARY_DIR} # Moc
)

add_executable(
        ${APPLICATION_NAME} MACOSX_BUNDLE # if cmdline app, no need of the BUNDLE 
        ${app_sources}
        ${app_headers}
        ${app_inlines}
        ${app_uis_moc}
        ${app_resources}
)
target_link_libraries(${APPLICATION_NAME} # target
        PUBLIC
        Radium::Core
        Radium::Engine
        Radium::GuiBase
        Radium::IO
        ${Qt5_LIBRARIES})

# optional for checking plugins compatibility wrt build mode
if (CMAKE_BUILD_TYPE MATCHES Debug)
    target_compile_definitions(${APPLICATION_NAME} PUBLIC APP_COMPILED_WITH_DEBUG_INFO)
endif ()

# let Radium configure all the installation procedure
configure_radium_app_install(
        NAME ${APPLICATION_NAME}
        USE_PLUGINS # optional, do  not set if app do not use Plugins
        RESOURCES ${PROJECT_SOURCE_DIR}/Shaders # optional : set of directories where resources are located
)

```

Reply:
For backup purpose only.
It's not compiled yet (as the EngineApp) feel free to update ;)

Reply:
> For backup purpose only.
> It's not compiled yet (as the EngineApp) feel free to update ;)

EngineApp is just commented out for CI as we have not yet find a way to test its execution during CI. But it will be compiled soon. So Prepare your app (I think it is welcome in the ExempleApps directory) to be compiled, even if you commented out the `include(primitivedemo)` cmake directive.

Reply:
resolved : I remove the old CMakeLists.txt.
When someone have time, he can create the cmake configuration to compile this app, and finish converting all draw primitives to the new mechanism, along with the test application.
And also add a unit test that shoot an image and compare to a reference.